### DISCUSSION TITLE
PPGW冷启动是否可以添加一个选项用关键词自动指定某个或者某几个节点？

### DISCUSSION THREAD

**Discussion by simon-yanghl** [kkkgo/PaoPaoGateWay#96] [问和答]
家中跑DNS和GW的小机器有点不稳，在琢磨自动定时冷启动或者搓个自动检测DNS的IP不可访问时自动断电重启的小玩意儿。
但是现在遇到一个问题，现有配置下，GW启动后默认是不选择节点的，如果选择自动检测速度最快的又会导致自动选择香港节点，但是香港节点又不能使用chatgpt。所以能否在ini配置中增加一个选项，启动后根据关键词自动选择节点。
比如说关键词是“新加坡01”或者模糊匹配“新加坡 01 xxx”，这样就可以重启后自动连接到节点

至于说自带规则，不好意思我没搞懂怎么弄，或者麻烦您指条明路应该如何下手，叩谢

**Comment by kkkgo** [comment]
你可以用ext_node功能排除香港节点。

**Reply by kkkgo** [reply]
GW一般不需要重启，除非是内存不足导致短暂卡死。如果替换了核心、开启net_rec等功能情况下可以适当增加内存分配。

**Comment by simon-yanghl** [comment]
> GW一般不需要重启，除非是内存不足导致短暂卡死。如果替换了核心、开启net_rec等功能情况下可以适当增加内存分配。

我是宿主机有问题，会死机，我要物理重启宿主机……

**Reply by simon-yanghl** [reply]
而且关键是我想用的其实只有美国/新加坡节点，如果是排除的话排除的太多了……

**Reply by kkkgo** [reply]
你可以使用`subconverter`项目进行更多的自定义过滤或者选择。如果手写规则，可以使用proxy-provider。

**Reply by kkkgo** [reply]
实际上你并不需要排除所有不使用的节点。ext_node功能是可以定义多个关键字进行排除。你可以调整你的test_node_url为对美国/新加坡友好的url，这样在节点可用的情况下，就会被优先选择，换句话说就是排除了比他们更快的测速url节点即可。

**Reply by simon-yanghl** [reply]
好的，我研究一下，看了几遍说明有点蒙，谢啦

**Reply by simon-yanghl** [reply]
我又重新看了一下相关说明，我的理解是找一个新加坡本地的url，这样测速的时候新加坡的ping是理论最低的，然后我再排除香港的关键字。结果就是极大概率选中新加坡的节点？

**Reply by kkkgo** [reply]
是的，这样即使你的新加坡节点不可用，也可以回退到其他节点。
